olmo/model.py

```diff
@@ -448,14 +448,15 @@ def __init__(self, layer_id: int, config: ModelConfig, cache: BufferCache):
             config.d_model, config.d_model, bias=config.include_bias, device=config.init_device
         )
 
-        # Feed-forward output projection.
-        self.ff_out = nn.Linear(
-            int(self.act.output_multiplier * self.hidden_size),
-            config.d_model,
-            bias=config.include_bias,
-            device=config.init_device,
-        )
-        self.ff_out._is_residual = True  # type: ignore
+        if self.config.block_type != BlockType.moe:
```

Comment:
Can you make this dependent on whether the block has a `ff_out`, instead of the block type?

Reply:
with `hasattr()`, I mean

Reply:
Do you mean `if hasattr(self, "ff_out"):`? Not sure that will work because the next lines are about creating self.ff_out so no block has it yet afaict